test/evergreen.yml

```diff
@@ -214,13 +214,18 @@ functions:
              SYSPYDIR=$(dirname $SYSPYLIB)
              NSYSPYLIB=$(ls $SYSPYDIR/lib/libpython*.dylib 2>/dev/null | head -1)
              if [ -f "$NSYSPYLIB" ]; then
-                 SYSPYLIB=$NSYSPYLIB
+               SYSPYLIB=$NSYSPYLIB
              fi
              if [ -d "$SYSPYDIR/Headers" ]; then
-                 SYSPYINCDEF="-DPYTHON_INCLUDE_DIR=$SYSPYDIR/Headers"
+               SYSPYINCDEF="$SYSPYDIR/Headers"
              fi
           fi
-          DEFINED_EVERGREEN_CONFIG_FLAGS="$DEFINED_EVERGREEN_CONFIG_FLAGS -DPYTHON_EXECUTABLE:FILEPATH=$SYSPY -DPYTHON_LIBRARY=$SYSPYLIB $SYSPYINCDEF"
+
+          if [ "${build_variant|}" = "macos-1100-arm64" ]; then
+            DEFINED_EVERGREEN_CONFIG_FLAGS="$DEFINED_EVERGREEN_CONFIG_FLAGS -DPython3_EXECUTABLE=$SYSPY -DPython3_LIBRARY=$SYSPYLIB -DPython3_INCLUDE_DIR=$SYSPYINCDEF"
+          else
+            DEFINED_EVERGREEN_CONFIG_FLAGS="$DEFINED_EVERGREEN_CONFIG_FLAGS -DPYTHON_EXECUTABLE:FILEPATH=$SYSPY -DPYTHON_LIBRARY=$SYSPYLIB -DPYTHON_INCLUDE_DIR=$SYSPYINCDEF"
+          fi
```

Comment:
Nice find! I was thinking we could create a variable specifically for the `-DPython3_EXECUTABLE=$SYSPY` configuration. For example :

```
          if [ "${build_variant|}" = "macos-1100-arm64" ]; then
            PYTHON_EXEC_FLAG="-DPython3_EXECUTABLE=$SYSPY"
          else
            PYTHON_EXEC_FLAG="-DPython3_EXECUTABLE:FILEPATH=$SYSPY"
          fi
          
          DEFINED_EVERGREEN_CONFIG_FLAGS="$DEFINED_EVERGREEN_CONFIG_FLAGS $PYTHON_EXEC_FLAG -DPYTHON_LIBRARY=$SYSPYLIB -DPYTHON_INCLUDE_DIR=$SYSPYINCDEF"
```

Let me know what you think.

Reply:
I tried that at one point. Unfortunately there are other differences - `PYTHON_LIBRARY` and `PYTHON_INCLUDE_DIR` need to be spelled differently across the versions.